bouncer/blacklist/signals.py

```diff
@@ -0,0 +1,30 @@
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from blacklist import models
+
+
+@receiver(pre_save, sender=models.IPEntry)
```

Comment:
Since all models have the same behavior, it would be best to:

* Group all signal handlers into one
* Remove the `sender` parameter
* Check if `instance` is of the correct class before proceeding